File: Boj.단계별/4_while문/javascript/1110.js
const fs = require('fs');
const n = Number(fs.readFileSync('/dev/stdin'));

let check_cycle = function (n,target,cnt){
    let new_num;
    if (n<10){
        new_num = n;
    }
    else{ new_num = n%10 + parseInt(n/10) };
    n = Number(String(n%10) + String(new_num%10))
    
    if (n===target){
        return cnt
    }
    cnt = check_cycle(n,target,cnt+1);
    return cnt
}
console.log(check_cycle(n,n,1));

File: Boj.단계별/4_while문/python/1110.py
num = target = int(input());
cnt = 0
while True:
    if num<10:
        new_num = num
    else:
        new_num = num%10 + num//10
    num = (num%10)*10 + new_num%10
    cnt += 1
    if num == target:
        break
print(cnt)    